libmario/mariolib/funcs.c

```diff
@@ -1004,6 +1004,7 @@ EXPORT void ADDCALL step_libmario(OSContPad *controllerData, s32 updateAnims) {
 
     surfacesUsed = 0;
 
+    gMarioState->input = INPUT_NONZERO_ANALOG;
```

Comment:
i don't think this should actually affect anything when tracing through the next line `execute_mario_action`

- `execute_mario_action` calls `update_mario_inputs` before performing any actions
- `update_mario_inputs` clears particles then runs `m->input = INPUT_NONE;`

afaict between this line you added and gMarioState->input getting cleared, there isn't any code running that would affect behaviors. was there something i didnt catch when looking through?